router/parse.js

```diff
@@ -16,77 +18,88 @@ export default function parse(uri) {
     items: []
   };
 
+  const parsers = [
+    ...whitelistRegex,
+    PANELS_REGEX
+  ];
+
   // Make sure we always have a trailing slash on the URI
   let nextUri = withTrailingSlash(uri);
 
-  let i = 0;
-
   do {
-    const parsed = new Url(nextUri);
+    let protocol = uri.match(PROTOCOL_REGEX)[1];
+    let app;
+    let path;
+    let fullPath;
+    let result;
+
+    for (let i = 0; i < parsers.length; i++) {
```

Comment:
What about using `find` instead of `for` with `break`? E.g.:

```
const parser = parsers.find(p => p.test(nextUri))
const result = parser.match(nextUri)
if (result) {
// ...
}
```
Thoughts?